ic-stable-structures/src/structure/stable_storage/unbounded.rs

```diff
@@ -62,6 +62,20 @@ where
         StableUnboundedIter(self.inner.iter().peekable())
     }
 
+    /// Returns number of chunks, used to store the entity.
+    pub fn chunks_number_of(&self, key: &K) -> Option<ChunkIndex> {
+        let first_chunk_key = Key::new(key);
+        let max_chunk_key = first_chunk_key.clone().with_max_chunk_index();
+
+        // first item in this iter will be the last chunk of the key.
+        self.inner
```

Comment:
How much expensive is this? It seems to me that this requires to decode the entire data

Reply:
It encodes the `key`, and performs one binary search in the map. The search requires reading (without decoding)  log_2(N) keys.